lib/propolis/src/enlightenment/bhyve.rs

```diff
@@ -0,0 +1,60 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Provides a bhyve-compatible guest-hypervisor interface.
+//!
+//! This interface supplies no special enlightenments; it merely identifies
+//! itself as a bhyve hypervisor in CPUID leaf 0x4000_0000.
+
+use cpuid_utils::{
+    bits::HYPERVISOR_BASE_LEAF, CpuidIdent, CpuidSet, CpuidValues,
+};
+
+use crate::{
+    accessors::MemAccessor,
+    common::{Lifecycle, VcpuId},
+    enlightenment::Enlightenment,
+    msr::{MsrId, RdmsrOutcome, WrmsrOutcome},
+};
+
+pub struct BhyveGuestInterface;
```

Comment:
This should probably get a comment about why it's here: just as a placeholder for the old CPUID leafs (providing whatever little value they hold), and nothing more.  This, given that there are no actual enlightenment features from the "bhyve brand" of hypervisor